src/meshes/structured_mesh_view.jl

```diff
@@ -0,0 +1,108 @@
+# By default, Julia/LLVM does not use fused multiply-add operations (FMAs).
+# Since these FMAs can increase the performance of many numerical algorithms,
+# we need to opt-in explicitly.
+# See https://ranocha.de/blog/Optimizing_EC_Trixi for further details.
+@muladd begin
+#! format: noindent
+
+"""
+    StructuredMeshView{NDIMS, RealT <: Real} <: AbstractMesh{NDIMS}
+
+A view on a structured curved mesh.
+"""
+mutable struct StructuredMeshView{NDIMS, RealT <: Real} <: AbstractMesh{NDIMS}
+    parent::StructuredMesh{NDIMS, RealT}
+    cells_per_dimension::NTuple{NDIMS, Int}
+    mapping::Any # Not relevant for performance
+    mapping_as_string::String
+    current_filename::String
+    index_min::NTuple{NDIMS, Int}
+    index_max::NTuple{NDIMS, Int}
+    unsaved_changes::Bool
+end
+
+"""
+    StructuredMeshView(parent; index_min, index_max)
+
+Create a StructuredMeshView on a StructuredMesh parent.
+
+# Arguments
+- `parent`: the parent StructuredMesh.
+- `index_min`: starting indices of the parent mesh.
+- `index_max`: ending indices of the parent mesh.
+"""
+function StructuredMeshView(parent::StructuredMesh{NDIMS, RealT};
+                            index_min = ntuple(_ -> 1, Val(NDIMS)),
+                            index_max = size(parent)) where {NDIMS, RealT}
+    @assert index_min <= index_max
+    @assert all(index_min .> 0)
+    @assert index_max <= size(parent)
+
+    # Calculate the domain boundaries.
+    cells_per_dimension = index_max .- index_min .+ 1
+    deltas = (parent.mapping.coordinates_max .- parent.mapping.coordinates_min) ./
+             parent.cells_per_dimension
+    coordinates_min = parent.mapping.coordinates_min .+ deltas .* (index_min .- 1)
+    coordinates_max = parent.mapping.coordinates_min .+ deltas .* index_max
+    mapping = coordinates2mapping(coordinates_min, coordinates_max)
+    mapping_as_string = """
+        coordinates_min = $coordinates_min
+        coordinates_max = $coordinates_max
+        mapping = coordinates2mapping(coordinates_min, coordinates_max)
+        """
+
+    return StructuredMeshView{NDIMS, RealT}(parent, cells_per_dimension, mapping,
+                                            mapping_as_string,
+                                            parent.current_filename,
+                                            index_min, index_max,
+                                            parent.unsaved_changes)
+end
+
+# Check if mesh is periodic
+function isperiodic(mesh::StructuredMeshView)
+    @unpack parent = mesh
+    return isperiodic(parent) && size(parent) == size(mesh)
+end
+
+function isperiodic(mesh::StructuredMeshView, dimension)
+    @unpack parent, index_min, index_max = mesh
+    return (isperiodic(parent, dimension) &&
+            index_min[dimension] == 1 &&
+            index_max[dimension] == size(parent, dimension))
+end
```

Comment:
Why do you check `size(parent) == size(mesh)` 

And why does `isperiodic(mesh::StructuredMeshView)` not call `isperiodic(mesh::StructuredMeshView, dimension)` for each dimension?

Reply:
For the meshview to be fully periodic, i.e. in x and y, it would need to cover the entire domain. This is why we check `size(parent) == size(mesh)`. In that case parent and meshview cover the same domain and meshview is periodic if the parent is. So, no need to check every dimension.